server/WebAPI/Controllers/AuthController.cs

```diff
@@ -13,25 +15,50 @@ public class AuthController : Controller
   {
     private readonly UserService _userService;
     private readonly AuthService _authService;
+    private readonly GoogleOauthService _googleOauthService;
 
-    public AuthController(UserService userService, AuthService authService)
+    public AuthController(UserService userService, AuthService authService, GoogleOauthService googleOauthService)
     {
       _userService = userService;
       _authService = authService;
+      _googleOauthService = googleOauthService;
     }
 
     [AllowAnonymous]
     [HttpPost]
     public IActionResult Auth(AuthRequest authRequest)
     {
       var user = _userService.GetUser(authRequest.Email, authRequest.Password);
-      if(user == null)
+      if (user == null)
         return Unauthorized();
 
       var token = _authService.MakeToken(user);
       return Ok(new AuthResponse { AccessToken = token });
     }
 
+    [AllowAnonymous]
+    [HttpPost("google")]
+    public async Task<IActionResult> AuthWithGoogle(GoogleOauthDTO gAuth) // I can transfer access code with path variable (google/{id})
+    {
+      string googleBearer = await _googleOauthService.GetToken(gAuth.Code);
+      string email = await _googleOauthService.GetEmail(googleBearer);
+      
+      var user = _userService.GetUser(email);
+      if (email == null || user != null && !user.IsOauth)
```

Comment:
what if legal user who registered with email (vasya@gmail.com) wants to log in with google (same email vasya@gmail.com)? do we prohibit this?

Reply:
In this case, yes. I'll fix it